Scripts/Helpers/ProxyUI.cs

```diff
@@ -0,0 +1,549 @@
+                                var shaderAlphaPropety = visibilityDefinition.alphaProperty;
+                                var materialsAndAssociatedColors = new List<Tuple<Material, Color, Color, Color, Color>>();
+                                // Material, original color, hidden color, animateFromColor(used by animating coroutines, not initialized here)
+                                foreach (var material in materialClones)
+                                {
+                                    // Clones that utilize the standard shader can be cloned and lose their enabled ZWrite value (1), if it was enabled on the material
+                                    // Set it again, to avoid ZWrite + transparency visual issues
+                                    if (visibilityType != VisibilityControlType.materialSwap && material.HasProperty(k_ZWritePropertyName))
+                                        material.SetFloat(k_ZWritePropertyName, 1);
+
+                                    Tuple<Material, Color, Color, Color, Color> materialAndAssociatedColors = null;
+                                    switch (visibilityDefinition.visibilityType)
+                                    {
+                                        case VisibilityControlType.colorProperty:
+                                            var originalColor = material.GetColor(visibilityDefinition.colorProperty);
+                                            materialAndAssociatedColors = new Tuple<Material, Color, Color, Color, Color>(material, originalColor, hiddenColor, Color.clear, Color.clear);
+                                            break;
+                                        case VisibilityControlType.alphaProperty:
+                                            var originalAlpha = material.GetFloat(shaderAlphaPropety);
+                                            var originalAlphaEncodedInColor = Color.white * originalAlpha;
+                                            // When animating based on alpha, use the Color.a value of the original, hidden, and animateFrom colors set below
+                                            materialAndAssociatedColors = new Tuple<Material, Color, Color, Color, Color>(material, originalAlphaEncodedInColor, hiddenAlphaEncodedInColor, Color.clear, Color.clear);
+                                            break;
+                                    }
+
+                                    materialsAndAssociatedColors.Add(materialAndAssociatedColors);
+                                }
+
+                                visibilityDefinition.materialsAndAssociatedColors = materialsAndAssociatedColors;
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (!affordancesDefinedInMap)
+                m_AffordanceMap.AffordanceDefinitions = generatedAffordanceDefinitions.ToArray();
```

Comment:
If there are affordances defined in the map, but not for every control, are those generated affordances being lost here?